python/pyspark/context.py

```diff
@@ -1255,6 +1255,16 @@ def getConf(self):
         conf.setAll(self._conf.getAll())
         return conf
 
+    def hadoopConfiguration(self):
+        """
+        Returns the Hadoop configuration used for the Hadoop code (e.g. file systems) we reuse.
+
+        As it will be reused in all Hadoop RDDs, it's better not to modify it unless you
+        plan to set some global configurations for all Hadoop RDDs.
+        Return :class:`Configuration` object
+        """
+        return self._jsc.hadoopConfiguration()
```

Comment:
It exposes a Py4J instance as is which we should avoid. Defining a Hadoop configuration class in PySpark doesn't make sense either. I am not sure which approach is the best. We could probably leave it and let advanced-users to use `sc. _jsc.hadoopConfiguration` for now.

Reply:
I had faced this issue when tried to setup s3a properties in pyspark job. The way how to do it was not present anywhere in the documentation but found a solution on [Stackoverflow.](https://stackoverflow.com/questions/28844631/how-to-set-hadoop-configuration-values-from-pyspark) Digging deeper I found a similar ticket opened in [Jira](https://issues.apache.org/jira/browse/SPARK-33436). That's why this PR was created. If we really do not need it, it can be closed. But in my mind, it is logical to have the same sc API I Scala and Python.

Reply:
If the point is to set the Hadoop configurations, you can add `spark.hadoop` prefix and set to Spark configurations.

Reply:
What the purpose of this [ticket](https://issues.apache.org/jira/browse/SPARK-33436) in this case?

Reply:
The purpose of the ticket is as [you described](https://github.com/apache/spark/pull/31768#discussion_r589082773). 

The story I relate in the ticket description is basically the same as yours: I wanted to set some Hadoop configs related to S3A; I couldn't find a direct way to do that; the best approach I could find went through `_.jsc`, so I thought that merited an improvement to PySpark so it had a direct means of doing the same thing.

Reply:
Ok, closing PR. I'd suggest abandoning the Jira ticket also, not to confuse people in the future.

Reply:
I'm not following why the ticket should be abandoned.

If there is an alternative solution for PySpark users who want to set S3A configs -- something that a) does not require them to go through `._jsc`, and b) is tested and works -- then please post a clear example of that on the ticket. That would serve as a useful reference to others.

Otherwise, I think the ticket is valid and should remain open.